client/src/components/form.tsx

```diff
@@ -8,35 +8,98 @@ import {
   Input,
   Spacer,
 } from "@chakra-ui/react";
-import { Field, FieldInputProps, Form, Formik } from "formik";
+import { Field, FieldInputProps, Form, Formik, FormikHelpers } from "formik";
 import { useContext } from "react";
 
 export default function InventoryForm() {
   const { dispatch } = useContext(InventoryContext);
 
-  const initialValues: { name?: string; quantities?: string } = {};
+  type User = {
+    username: string;
+    firstName: string;
+    lastName: string;
+  }
 
+  type Values = { name?: string; quantities?: string; username?: string };
+
+  const users: User[] = [
+    {
+      username: 'jdoe',
+      firstName: 'John',
+      lastName: 'Doe',
+    },
+    {
+      username: 'ssmith',
+      firstName: 'Sarah',
+      lastName: 'Smith',
+    },
+    {
+      username: 'asimons',
+      firstName: 'Amanda',
+      lastName: 'Simons',
+    }
+  ]
+
+  const initialValues: Values  = {
+    username: users[0].username
+  };
+
+  // this helper (onlyContainsNumbers) should even reject floats/doubles
+  const onlyContainsNumbers = (str: string) => /^\d+$/.test(str);
+  const isNumber = (str: string) => ! isNaN(Number(str)) && onlyContainsNumbers(str);
+
+  const validateQuantities = (quantities: string): boolean => {
+    const splitted = quantities.split(',')
+    for (let i = 0; i < splitted.length; i++) {
+      const number = splitted[i];
+      if (!isNumber(number.trim())) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  const formSubmitted = ((values: Values, actions: FormikHelpers<Values>) => {
+    if (!values || !values.name || !values.username || !values.quantities) {
+      alert('Please ensure all values are populated!')
+      actions.setSubmitting(false);
+      return;
+    }
+
+    if (!validateQuantities(values.quantities)) {
+      alert('Please ensure quantities are only comma-separated numbers!');
+      actions.setSubmitting(false);
+      return;
+    }
+
+    fetch(`http://localhost:3001/inventory/${values.name}`, {
```

Comment:
I should be loading the API URL from an environment variable instead of it being hardcoded.